### DISCUSSION TITLE
Migrate from v3 to v5 in b4a code

### DISCUSSION THREAD

**Discussion by cas6678** [mapsforge/mapsforge#1383] [Q&A] [answered]
Hello everyone.

For a few years I have a project in B4A that uses OSMDroid308 libraries and V3 maps. I see now that only version 5 (v5) maps are available. But these do not work with my code.

Can you explain the differences between v3 and v5 and where should I start to restructure my code? New libraries? Other methods? some example?

Thank you.

**Comment by devemux86** [comment, marked as answer]
It's not just the maps, the library has changed as well.

See the [getting started example](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/GettingStarted.java) on how to use the Mapsforge API.
(and the other examples in [mapsforge-samples-android](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-android))

**Reply by cas6678** [reply]
Thank you very much for your answer.

I have seen the links that you indicate but it is too much for me. :(

I understand that perhaps there are still no libraries in b4a to be able to operate with this new way?

**Reply by devemux86** [reply]
I can not know that, better ask their maintainers.

If you want to use directly the Mapsforge API, see the [integration guide](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md).

**Reply by cas6678** [reply]
thanks @devemux86
